addons/folo/common/src/main/java/org/commonjava/indy/folo/data/FoloRecordCassandra.java

```diff
@@ -0,0 +1,426 @@
+        if(one != null) {
+            Boolean sealed = one.getBool("sealed");
+            if(!sealed) {
+                return true;
+            } else {
+                return false;
+            }
+        }else {
+            return false;
+        }
+    }
+
+    @Override
+    public TrackedContent get(TrackingKey key) {
+        List<DtxTrackingRecord> trackingRecords =  getDtxTrackingRecordsFromDb(key);
+        return transformDtxTrackingRecordToTrackingContent(key,trackingRecords);
+    }
+
+    @Override
+    public TrackedContent seal(TrackingKey trackingKey) {
+        List<DtxTrackingRecord> trackingRecords =  getDtxTrackingRecordsFromDb(trackingKey);
+        DtxTrackingRecord recordCheck = trackingRecords.get(0);
+        if(recordCheck.getState()) {
+            logger.debug( "Tracking record: {} already sealed! Returning sealed record.", trackingKey );
+            return transformDtxTrackingRecordToTrackingContent(trackingKey,trackingRecords);
+        }
+        logger.debug( "Sealing record for: {}", trackingKey );
+        for(DtxTrackingRecord record : trackingRecords) {
+            record.setState(true);
+            trackingMapper.save(record);
+        }
+        return transformDtxTrackingRecordToTrackingContent(trackingKey,trackingRecords);
+    }
+
+    @Override
+    public Set<TrackingKey> getInProgressTrackingKey() {
+        BoundStatement inProgress = getTrackingRecordBySealed.bind(false);
+        ResultSet getInProgressRecords = session.execute(inProgress);
+        List<Row> allInProgress = getInProgressRecords.all();
+        Iterator<Row> iterator = allInProgress.iterator();
+
+        Set<TrackingKey> trackingKeys = new HashSet<>();
+        while (iterator.hasNext()) {
+            Row next = iterator.next();
+            String tracking_key = next.getString("tracking_key");
+            trackingKeys.add(new TrackingKey(tracking_key));
+        }
+        return trackingKeys;
+    }
+
+    @Override
+    public Set<TrackingKey> getSealedTrackingKey() {
+        BoundStatement inProgress = getTrackingRecordBySealed.bind(true);
+        ResultSet getInProgressRecords = session.execute(inProgress);
+        List<Row> allInProgress = getInProgressRecords.all();
+        Iterator<Row> iterator = allInProgress.iterator();
+
+        Set<TrackingKey> trackingKeys = new HashSet<>();
+        while (iterator.hasNext()) {
+            Row next = iterator.next();
+            String tracking_key = next.getString("tracking_key");
+            trackingKeys.add(new TrackingKey(tracking_key));
+        }
+
+        return trackingKeys;
+    }
+
+    @Override
+    public Set<TrackedContent> getSealed() {
+
+        Set<TrackedContent> trackedContents = new HashSet<>();
+        Set<TrackingKey> sealedTrackingKeys = getSealedTrackingKey();
+
+        for(TrackingKey trackingKey : sealedTrackingKeys) {
+
+            List<DtxTrackingRecord> dtxTrackingRecordsFromDb = getDtxTrackingRecordsFromDb(trackingKey);
```

Comment:
We may need some mechanism for expanding the maximum query results for this call, since this is effectively exporting our entire database of tracking record entries, or at least the subset that got sealed. In any case, it'll be a **very** inefficient query and return a **lot** of results.

Eventually, we need to eliminate this to some sort of callback-driven process, or we'll never escape the huge latencies + system saturation that this kind of thing can produce.

Reply:
> We may need some mechanism for expanding the maximum query results for this call, since this is effectively exporting our entire database of tracking record entries, or at least the subset that got sealed. In any case, it'll be a very inefficient query and return a lot of results.

Because of created indexing on 'sealed' record we will have best optimization that cassandra could provide for this type of query. Once again here is cassandra documentation:
"An index provides a means to access data in Cassandra using attributes other than the partition key. The benefit is fast, efficient lookup of data matching a given condition. The index indexes column values in a separate, hidden table from the one that contains the values being indexed."

BTW this is query for returning all sealed records, if we don't follow cassandra. best practices then every other type of implementation. will. be lot more. inefficient then this one provided as. suggestion from their. documentation.